src/decls/types.js

```diff
@@ -0,0 +1,54 @@
+// @flow
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import SpanContext from '../span_context.js';
+
+declare type Endpoint = {
+    ipv4: number,
+    port: number,
+    serviceName: string
+};
+
+declare type Annotation = {
+    timestamp: number,
+    value: string,
+    host: ?Endpoint
+};
+
+declare type BinaryAnnotation = {
+    key: string,
+    value: any,
+    annotationType: string,
+    host: ?Endpoint
+};
+
+declare type Reference = {
+    type(): void;
+    referencedContext(): SpanContext;
+};
+
+declare type spanFields = {
```

Comment:
Its not the span structure.  I think the naming is bad, but its really the parameters to start Span, since opentracing's start span method called their parameters fieldsOrName (but they guarantee you the fields)  I used fields as the name, so then I called this type spanFields.  Maybe "startSpanFields"?